async.go

```diff
@@ -166,3 +133,48 @@ func (sr *snowflakeRestful) getAsync(
 	}
 	return nil
 }
+
+func getQueryResultWithRetriesForAsyncMode(
+	ctx context.Context,
+	sr *snowflakeRestful,
+	URL *url.URL,
+	headers map[string]string,
+	timeout time.Duration) (*execResponse, error) {
+	var respd *execResponse
+	retry := 0
+	retryPattern := []int32{1, 1, 2, 3, 4, 8, 10}
+
+	for {
+		resp, err := sr.FuncGet(ctx, sr, URL, headers, timeout)
+		if err != nil {
+			logger.WithContext(ctx).Errorf("failed to get response. err: %v", err)
+			return respd, err
+		}
+		defer resp.Body.Close()
+
+		respd = &execResponse{} // reset the response
+		err = json.NewDecoder(resp.Body).Decode(&respd)
+		if err != nil {
+			logger.WithContext(ctx).Errorf("failed to decode JSON. err: %v", err)
+			return respd, err
+		}
+		if respd.Code != queryInProgressAsyncCode {
```

Comment:
Added logging for retry counter.

Reply:
But in a way you did it - after reaching retry pattern length, the same counter is logged again and again? Maybe it would be better to increment it always and when reaching the retryPattern prune it to the length?

Reply:
Updated.